### ISSUE TITLE
NaN is useless, user-unfriendly and should never be displayed in the UI

### ISSUE THREAD

**Issue by LeaVerou** [mavoweb/mavo#219]
Instead, it should be treated as 0. In most cases, that's the intention anyway. 
We do this for `null` already.
Thoughts?

**Comment by DmitrySharabin** [comment]
I might be mistaken, but I believe that `null` and `NaN` are a bit different: `null` represents the absence of any value, whereas `NaN`, in most cases, shows that an error occurred.

What if we instead of number `0`, that looks like a suitable result (and an end-user won't see an error), provide a simple hint which shows that something went wrong?

E.g., Excel provides a couple of simple techniques: [a simple error message](https://www.exceltactics.com/definitive-guide-excel-error-types-error-handling/), [a green triangle, and a yellow diamond](https://www.lifewire.com/thmb/6UaZIT7TS3DIcAFnWa35DmcPU6k=/1320x0/filters:no_upscale():max_bytes(150000):strip_icc():format(webp)/ErrorDropdown-5bee099e46e0fb00267bb396.jpg). To my mind, users get used to it. Google Sheets uses [a similar approach](https://infoinspired.com/wp-content/uploads/2018/02/VALUE-error-in-Google-Sheets.jpg).

Or, if we have a look at Word, it uses colored squiggly lines to highlight errors.

I don't know what the best practices, in this case, are. Just shared my thoughts. 😊

**Comment by LeaVerou** [comment]
You are not mistaken, `NaN` does serve a distinct purpose. However, it also means nothing to novices, and even less to their users. `0` is a meaningful result, I don't think it would be good to erroneously report 0 when the result is invalid.
The spreadsheet approaches are good, but they would mean we'd need to inject more HTML and styling into users' apps, which may result in weird rendering (and is unclear how to do in some cases, e.g. attributes)
